packages/components/codemods/utils/updateKaioImports.ts

```diff
@@ -0,0 +1,233 @@
+type ImportsToRemoveMap = Map<string, ImportsToRemove>
+// Key is module specifier (eg. "@kaizen/components")
+type ImportsToAddMap = Map<string, ImportsToAdd>
+
+export type UpdateKaioImportsArgs = {
+  importsToRemove?: ImportsToRemoveMap
+  importsToAdd?: ImportsToAddMap
+}
+
+export const updateKaioImports =
+  ({ importsToRemove, importsToAdd }: UpdateKaioImportsArgs) =>
+  (context: ts.TransformationContext) =>
+  (rootNode: ts.Node): ts.Node => {
+    if (!ts.isSourceFile(rootNode)) return rootNode
+
+    if (!importsToRemove && !importsToAdd) return rootNode
+
+    const { factory } = context
+
+    const statements = Array.from(rootNode.statements)
+
+    if (importsToRemove) {
+      Array.from(importsToRemove.keys()).forEach(moduleSpecifier => {
+        const importIndex = statements.findIndex(
+          s =>
+            ts.isImportDeclaration(s) &&
+            (s.moduleSpecifier as ts.StringLiteral).text === moduleSpecifier
+        )
+
+        if (importIndex === -1) return
+
+        const importDeclaration = statements[
+          importIndex
+        ] as ts.ImportDeclaration
+
+        const updatedImportDeclaration = removeNamedImports(
+          factory,
+          importDeclaration,
+          importsToRemove.get(moduleSpecifier)!
+        )
+
+        if (updatedImportDeclaration === null) {
+          // Remove import statement as there are no more imports
+          statements.splice(importIndex, 1)
+          return
+        }
+
+        // Update import statement
+        statements[importIndex] = updatedImportDeclaration
+      })
+    }
+
+    if (importsToAdd) {
+      Array.from(importsToAdd.keys()).forEach(newModuleSpecifier => {
+        const importIndex = statements.findIndex(
+          s =>
+            ts.isImportDeclaration(s) &&
+            (s.moduleSpecifier as ts.StringLiteral).text === newModuleSpecifier
+        )
+
+        if (importIndex === -1) {
+          const fallbackKaioImportIdx = statements.findIndex(
+            s =>
+              ts.isImportDeclaration(s) &&
+              (s.moduleSpecifier as ts.StringLiteral).text.includes(
+                "@kaizen/components"
+              )
+          )
+
+          const newImport = createImportDeclaration(
+            factory,
+            importsToAdd.get(newModuleSpecifier)!,
+            newModuleSpecifier
+          )
+          statements.splice(fallbackKaioImportIdx + 1, 0, newImport)
+          return
+        }
+
+        const importDeclaration = statements[
+          importIndex
+        ] as ts.ImportDeclaration
+
+        const updatedImportDeclaration = updateNamedImports(
+          factory,
+          importDeclaration,
+          importsToAdd.get(newModuleSpecifier)!
+        )
+
+        // Update import statement
+        statements[importIndex] = updatedImportDeclaration
+      })
+    }
+
+    return factory.updateSourceFile(rootNode, statements)
+  }
+
+/* Consumer helpers */
+export const setImportToRemove = (
```

Comment:
I kind of understand these in the abstract by looking at how they're used in the upgradeIconV1 but can we add a little more context in the code comments, potentially with an example?